DSharpPlus/Entities/Guild/DiscordMember.cs

```diff
@@ -61,11 +62,24 @@ internal DiscordMember(TransportMember mbr)
             this.Nickname = mbr.Nickname;
             this.PremiumSince = mbr.PremiumSince;
             this.IsPending = mbr.IsPending;
-
+            this._avatarHash = mbr.AvatarHash;
             this._role_ids = mbr.Roles ?? new List<ulong>();
             this._role_ids_lazy = new Lazy<IReadOnlyList<ulong>>(() => new ReadOnlyCollection<ulong>(this._role_ids));
         }
 
+        /// <summary>
+        /// Gets the member's avatar for the current guild.
+        /// </summary>
+        public string GuildAvatarHash => this._avatarHash ?? this.User.AvatarHash;
+
+        /// <summary>
+        /// Gets the members avatar url for the current guild.
+        /// </summary>
+        public string GuildAvatarUrl
+            => !string.IsNullOrWhiteSpace(this.GuildAvatarHash) ? (this.GuildAvatarHash.StartsWith("a_") ? $"https://cdn.discordapp.com/guilds/{this._guild_id}/users/{this.Id}/avatars/{this.GuildAvatarHash}.gif?size=1024" : $"https://cdn.discordapp.com/guilds/{this._guild_id}/users/{this.Id}/avatars/{this.GuildAvatarHash}.png?size=1024") : this.DefaultAvatarUrl;
```

Comment:
`[JsonIgnore]` should be added to these as well. I don't think `DiscordMember` is ever actually serialized but do it for uniformity I suppose.

Reply:
> I don't think DiscordMember is ever actually serialized but do it for uniformity I suppose.

'tis not, but `Ignore` nonetheless if applicable.